### PR TITLE
Implement `tables.delete` rpc endpoint

### LINKED ISSUE #3607: Implement `tables.delete` RPC method

### PR DESCRIPTION
**Pull request by Anish9901**
Fixes #3607

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: db/sql/00_msar.sql
Comment (Anish9901) [reply db/sql/00_msar.sql:2152]: I am okay with returning `NULL` here instead of the table name. There are a couple of reasons why I chose to make this change.
- Our `columns.delete` function returns the number of columns deleted. 
- Earlier this function used to return the SQL query that was being run on the db for dropping the table.

Keeping this in mind I thought if an end user were to use our RPC endpoint, it might be beneficial for them if they receive some kind of a not null response.

File: db/sql/00_msar.sql
Comment (Anish9901) [inline db/sql/00_msar.sql:2138]: Rethinking about this made me realize that we could allow `if_exists` with this function if we change the datatype of tab_id from `oid` -> `regclass` and we can document in the python function's docstring that `if_exists` only works when the `tab_id` is a fully qualified table name. I'd like to know your thoughts on this @mathemancer, @seancolsen.

File: db/tables/operations/drop.py
Comment (seancolsen) [inline db/tables/operations/drop.py:8]: Sorry — as I was looking at this code again when implementing schemas.delete, I noticed another nitpick I wanted to mention...

We had `drop_table` and then you added `drop_table_from_schema`. These function names, when taken together are somewhat confusing. I propose renaming them as follows:

- `drop_table` → `drop_table_via_name`
- `drop_table_from_schema` → `drop_table_via_oid`

File: db/sql/00_msar.sql
Comment (mathemancer) [reply db/sql/00_msar.sql:2152]: I prefer returning either `1` for "1 table dropped", or the name, is preferable:
- Even if we're not displaying on the FE, it's useful for a user just calling the endpoint.
- It distinguishes between the "no table found" and "found and dropped table" cases, whenever `if_exists=true`

File: db/tables/operations/drop.py
Comment (mathemancer) [reply db/tables/operations/drop.py:8]: I prefer `drop_table_from_database`. Keep in mind, the `via_name` function will be removed when we get rid of the relevant REST endpoint and model.

File: db/sql/00_msar.sql
Comment (mathemancer) [reply db/sql/00_msar.sql:2138]: At the SQL level, I prefer keeping the flag for the `schema, tab_name` version, and removing it from the `tab_oid` version where it makes no sense (without custom exception handling). For the API level, I prefer defaulting to returning a success marker (e.g., the table name) if the table actually existed and was dropped, and a `null` if nothing happened (because the table didn't exist).

**Review by mathemancer — COMMENTED** [review]
After chatting with @Anish9901, I think we should block this PR until the `__msar.get_relation_name(oid)` function is updated to return `null` if no such relation exists.

**Comment by Anish9901** [comment]
@mathemancer I realized that we already have a function named `msar.get_relation_name_or_null(oid)`, I've refactored the drop_table function to use that instead of refactoring `__msar.get_relation_name(oid)`. The PR should be unblocked and ready for re-review.

File: db/tables/operations/drop.py
Comment (Anish9901) [reply db/tables/operations/drop.py]: done!

File: db/sql/00_msar.sql
Comment (seancolsen) [reply db/sql/00_msar.sql:2152]: I thought we agreed yesterday that we would not bother with using parameters for `if_exists`. Instead we would use queries that drop things without `IF EXISTS` (raising an exception if the thing is not found) and add tables without `IF NOT EXISTS` (raising an exception if the thing is already there). That's simpler.

> it's useful for a user just calling the endpoint

My point here is that we don't currently have any use cases like this. And I don't foresee having any use cases like this. If we do end up having these use cases then we can add this complexity at that point.

Returning void is simpler. Changing a void return to a value return is a non-breaking change. Easy to make later on! Changing a value return to a different return is a breaking change. Harder to make later on! I want to start simple and then grow the complexity only as needed.

File: mathesar/rpc/tables.py
Comment (seancolsen) [reply mathesar/rpc/tables.py]: Ok, @mathemancer that's definitely different from how I tend to structure code. But I don't have a strong opinion on it. I'll defer to you. It might be nice to document a code standard about this at some point.

**Review by mathemancer — APPROVED** [review]
This seems to work, I think we should merge.

**Comment by mathemancer** [comment]
I've approved this, but I'll wait to merge until after our check-in later today.

**Pull request merged**
